### PR TITLE
fix(dav): remove unused userId param of some ocs routes

### PR DESCRIPTION
**Pull request by st3iny**
* Resolves: # 

## Summary

We only allow to set the data for the requesting user anyway. AFAIK, this API is not yet used anywhere outside of personal settings.

### Proof

https://github.com/nextcloud/server/blob/9a206c6282a55c9fdeb99abe8684199c0f934d7f/apps/dav/lib/Controller/OutOfOfficeController.php#L130-L135

https://github.com/nextcloud/server/blob/9a206c6282a55c9fdeb99abe8684199c0f934d7f/apps/dav/lib/Controller/OutOfOfficeController.php#L175

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by provokateurin — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Comment by nickvergessen** [comment]
This was intentional to:
1. be able to identify calls in access.log more easily
2. not requiring a new route when implementing admin/group lead support with https://github.com/nextcloud/server/issues/40397#issuecomment-1833509344

File: apps/dav/src/components/AbsenceForm.vue [hunk 1]
```diff
@@ -117,7 +117,7 @@ export default {
 
 			this.loading = true
 			try {
-				await axios.post(generateOcsUrl('/apps/dav/api/v1/outOfOffice/{userId}', { userId: getCurrentUser().uid }), {
```
Comment (nickvergessen) [inline apps/dav/src/components/AbsenceForm.vue:120]: I think I added the function import too, so if we stay with removing it, also should remove the import.

Comment (st3iny) [reply apps/dav/src/components/AbsenceForm.vue:120]: Thanks for the hint. It seems to be unused so I remove it now.

**Comment by provokateurin** [comment]
> not requiring a new route when implementing admin/group lead support with

This could just be a query parameter, no? Then we don't need an additional endpoint later.

**Comment by nickvergessen** [comment]
> Conflicting files
dist/dav-settings-personal-availability.js
dist/dav-settings-personal-availability.js.map 

If it's not merged before rc3 tonight it will be in 28 final, it's ocs API => 3 years deprecation
So we keep it now?

**Comment by blizzz** [comment]
>  So we keep it now?

Yes, no stunts five minutes before twelve anymore.

**Pull request closed without merging**